topics/docker.md

```diff
@@ -1,270 +1,100 @@
-It packages all the output classes, resources, and all the required dependencies into a single JAR file,
-and appends a manifest file to tell Java which is the entry-point main class containing the main method. 
-
-First, you need to add the shadow plugin dependency in your `build.gradle` file:
-
-```groovy 
-buildscript {
-    ...
-    repositories {
-        ...
-        maven { url "https://plugins.gradle.org/m2/" }
-    }
-    dependencies {
-        ...
-        classpath "com.github.jengelman.gradle.plugins:shadow:2.0.1"
-    }
-}
-```
-
-After that, you have to apply it, along with the application plugin:
-
-```groovy
-apply plugin: "com.github.johnrengelman.shadow"
-apply plugin: 'application'
-``` 
-
-Then specify the main class, so it knows what to run when running the java's JAR inside Docker:
-
-```groovy
-mainClassName = 'org.sample.ApplicationKt'
-```
-
-The string is the fully qualified name of the class containing your `main` function. When `main` function is a top-level
-function in a file, the class name is the file name with the `Kt` suffix. In the example above, `main` function is in the
-file `Application.kt` in package `org.sample`.
-
-Finally, you have to configure the shadow plugin:
-
-```groovy
-shadowJar {
-    baseName = 'my-application'
-    classifier = null
-    version = null
-}
-```
-
-Now you can run `./gradlew build` to build and package your application.
-You should get `my-application.jar` in `build/libs` folder.  
-
-For more information about configuring this plugin see [documentation for the plugin](http://imperceptiblethoughts.com/shadow/)
-
-So a full `build.gradle` file would look like this:
-
 <tabs>
 
 <tab title="Gradle">
 
 ```groovy
-buildscript {
-    ext.kotlin_version = '%kotlin_version%'
-    ext.ktor_version = '%ktor_version%'
-    ext.logback_version = '1.2.3'
-    ext.slf4j_version = '1.7.25'
-    repositories {
-        mavenCentral()
-        maven { url "https://plugins.gradle.org/m2/" }
-    }
-    dependencies {
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
-        classpath "com.github.jengelman.gradle.plugins:shadow:2.0.1"
-    }
-}
-
-apply plugin: 'kotlin'
-apply plugin: "com.github.johnrengelman.shadow"
-apply plugin: 'application'
-
-mainClassName = "io.ktor.server.netty.EngineMain"
-
-sourceSets {
-    main.kotlin.srcDirs = [ 'src' ]
-    main.resources.srcDirs = [ 'resources' ]
-}
-
-repositories {
-    mavenCentral()
-}
-
-dependencies {
-    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk8:$kotlin_version"
-    compile "io.ktor:ktor-server-netty:$ktor_version"
-    compile "io.ktor:ktor-html-builder:$ktor_version"
-    compile "ch.qos.logback:logback-classic:$logback_version"
-}
-
-kotlin.experimental.coroutines = 'enable'
-
-shadowJar {
-    baseName = 'my-application'
-    classifier = null
-    version = null
-}
 ```
+{src="docker/build.gradle"}
```

Comment:
Paths to snippets should start from the `snippets` directory: 
```
{src="snippets/docker/build.gradle"}
```

Reply:
My bad. Apologies (on this and the other ones).